tritium-caffeine/src/main/java/com/palantir/tritium/metrics/caffeine/CacheStats.java

```diff
@@ -95,22 +125,25 @@ public void recordMisses(@NonNegative int count) {
     @Override
     public void recordLoadSuccess(@NonNegative long loadTime) {
         loadSuccessTimer.update(loadTime, TimeUnit.NANOSECONDS);
-        totalLoadNanos.add(loadTime);
+        totalLoadTime.add(loadTime);
     }
 
     @Override
     public void recordLoadFailure(@NonNegative long loadTime) {
         loadFailureTimer.update(loadTime, TimeUnit.NANOSECONDS);
-        totalLoadNanos.add(loadTime);
+        totalLoadTime.add(loadTime);
     }
 
     @Override
     public void recordEviction(@NonNegative int weight, RemovalCause cause) {
-        Meter counter = evictionMeters.get(cause);
-        if (counter != null) {
-            counter.mark(weight);
+        Meter evictionMeter = evictionMeters.get(cause);
+        if (evictionMeter != null) {
+            evictionMeter.mark();
+        }
+        Meter evictionWeightMeter = evictionWeightMeters.get(cause);
+        if (evictionWeightMeter != null) {
+            evictionWeightMeter.mark(weight);
         }
-        evictionsTotalMeter.mark(weight);
```

Comment:
This was incorrect. The `evictionCount` provided to `CacheStats.of` should be the total number of evictions, not the total eviction weight. See the default `StatsCounter` for an example:

https://github.com/ben-manes/caffeine/blob/5b8e837b33949d1527eb9f76653725e2a53454ac/caffeine/src/main/java/com/github/benmanes/caffeine/cache/stats/ConcurrentStatsCounter.java#L74-L79

I've update this metric to record both the eviction count and eviction weight per-cause.